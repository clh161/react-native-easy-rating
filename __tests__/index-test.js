// __tests__/Intro-test.js
import React from 'react';
import renderer from 'react-test-renderer';
import Rating from '../index';

test('Render', () => {
  const tree = renderer.create(
    <Rating
      max={5}
      iconWidth={24}
      iconHeight={24}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render Rating 3', () => {
  const tree = renderer.create(
    <Rating
      rating={3}
      max={5}
      iconWidth={24}
      iconHeight={24}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render Max 10', () => {
  const tree = renderer.create(
    <Rating
      max={10}
      iconWidth={24}
      iconHeight={24}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render Icon Size 64', () => {
  const tree = renderer.create(
    <Rating
      max={5}
      iconWidth={64}
      iconHeight={64}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
