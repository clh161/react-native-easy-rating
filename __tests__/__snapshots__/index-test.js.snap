// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "flexDirection": "row",
      },
    ]
  }
>
  <Image
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../images/icon_star_unselected.png",
      }
    }
    style={
      Object {
        "height": 24,
        "width": 24,
      }
    }
  />
  <Image
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../images/icon_star_unselected.png",
      }
    }
    style={
      Object {
        "height": 24,
        "width": 24,
      }
    }
  />
  <Image
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../images/icon_star_unselected.png",
      }
    }
    style={
      Object {
        "height": 24,
        "width": 24,
      }
    }
  />
  <Image
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../images/icon_star_unselected.png",
      }
    }
    style={
      Object {
        "height": 24,
        "width": 24,
      }
    }
  />
  <Image
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../images/icon_star_unselected.png",
      }
    }
    style={
      Object {
        "height": 24,
        "width": 24,
      }
    }
  />
</View>
`;

exports[`Render Icon Size 64 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "flexDirection": "row",
      },
    ]
  }
>
  <Image
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../images/icon_star_unselected.png",
      }
    }
    style={
      Object {
        "height": 64,
        "width": 64,
      }
    }
  />
  <Image
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../images/icon_star_unselected.png",
      }
    }
    style={
      Object {
        "height": 64,
        "width": 64,
      }
    }
  />
  <Image
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../images/icon_star_unselected.png",
      }
    }
    style={
      Object {
        "height": 64,
        "width": 64,
      }
    }
  />
  <Image
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../images/icon_star_unselected.png",
      }
    }
    style={
      Object {
        "height": 64,
        "width": 64,
      }
    }
  />
  <Image
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../images/icon_star_unselected.png",
      }
    }
    style={
      Object {
        "height": 64,
        "width": 64,
      }
    }
  />
</View>
`;

exports[`Render Max 10 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "flexDirection": "row",
      },
    ]
  }
>
  <Image
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../images/icon_star_unselected.png",
      }
    }
    style={
      Object {
        "height": 24,
        "width": 24,
      }
    }
  />
  <Image
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../images/icon_star_unselected.png",
      }
    }
    style={
      Object {
        "height": 24,
        "width": 24,
      }
    }
  />
  <Image
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../images/icon_star_unselected.png",
      }
    }
    style={
      Object {
        "height": 24,
        "width": 24,
      }
    }
  />
  <Image
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../images/icon_star_unselected.png",
      }
    }
    style={
      Object {
        "height": 24,
        "width": 24,
      }
    }
  />
  <Image
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../images/icon_star_unselected.png",
      }
    }
    style={
      Object {
        "height": 24,
        "width": 24,
      }
    }
  />
  <Image
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../images/icon_star_unselected.png",
      }
    }
    style={
      Object {
        "height": 24,
        "width": 24,
      }
    }
  />
  <Image
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../images/icon_star_unselected.png",
      }
    }
    style={
      Object {
        "height": 24,
        "width": 24,
      }
    }
  />
  <Image
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../images/icon_star_unselected.png",
      }
    }
    style={
      Object {
        "height": 24,
        "width": 24,
      }
    }
  />
  <Image
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../images/icon_star_unselected.png",
      }
    }
    style={
      Object {
        "height": 24,
        "width": 24,
      }
    }
  />
  <Image
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../images/icon_star_unselected.png",
      }
    }
    style={
      Object {
        "height": 24,
        "width": 24,
      }
    }
  />
</View>
`;

exports[`Render Rating 3 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "flexDirection": "row",
      },
    ]
  }
>
  <Image
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../images/icon_star_selected.png",
      }
    }
    style={
      Object {
        "height": 24,
        "width": 24,
      }
    }
  />
  <Image
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../images/icon_star_selected.png",
      }
    }
    style={
      Object {
        "height": 24,
        "width": 24,
      }
    }
  />
  <Image
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../images/icon_star_selected.png",
      }
    }
    style={
      Object {
        "height": 24,
        "width": 24,
      }
    }
  />
  <Image
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../images/icon_star_unselected.png",
      }
    }
    style={
      Object {
        "height": 24,
        "width": 24,
      }
    }
  />
  <Image
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../images/icon_star_unselected.png",
      }
    }
    style={
      Object {
        "height": 24,
        "width": 24,
      }
    }
  />
</View>
`;
